### ISSUE TITLE
if envPath is set but does not exist, loading tests will fail

### ISSUE THREAD

**Issue by alexkli** [hbenl/vscode-mocha-test-adapter#108]
**Problem:** If `mochaExplorer.envPath` is set in the `settings.json`, but the file there does not exist, then it will fail to load tests in the tree and no tests can be run.

**Expected:** It should ignore it if the file does not exist.

I have a couple projects with `.env` files in their root, that are often needed in tests. So I have this in my global `settings.json` (I almost never have a local settings.json). But once I set it, the projects without an `.env` file will fail to load tests.

**Details**

Version: `2.3.0`

Example config:

```
"mochaExplorer.envPath": ".env"
```

Then without an `.env`. in the workspace root, test explorer side bar will show "Error while loading tests - click to show" and clicking on it shows this error:

```
[Error: ENOENT: no such file or directory, open '/path/to/my/workspace/.env'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/path/to/my/workspace/.env'
}
```

**Comment by alexkli** [comment]
Looks like this line here needs to be guarded by a file existence check: https://github.com/hbenl/vscode-mocha-test-adapter/blob/7055912ddff772a76f48a08876ee8e76273f5bb4/src/configReader.ts#L363

**Comment by hbenl** [comment]
Actually I'd like to keep it this way: I'm not a fan of software that tries to be smarter than its user. If the configuration contains an error (e.g. a typo in the filename), the extension should alert the user to this rather than ignoring it, so that the user doesn't have to guess why his `.env` file has no effect. And I don't think writing a log message (which most users won't see) is good enough to alert the user.
If you want to set `mochaExplorer.envPath` in your user settings, you'll have to override it by adding `"mochaExplorer.envPath": null` in those projects that don't have a `.env` file.

**Comment by alexkli** [comment]
@hbenl I think there might be a misunderstanding: I have this set in my global **user settings**. Not in the project's workspace settings. I agree that in the workspace setting it should fail. But if set in the user settings, it should be liberal in not failing if it does not exist (it could just log a warning).

I rarely use workspace settings (so vscode/settings.json does not exist, not gitignore setup etc.) so I prefer having configurations in the global user settings.
> you'll have to override it by adding "mochaExplorer.envPath": null in those projects that don't have a .env file.

Unfortunately, that is completely impractical, since most projects (and foreign projects that I clone and build) don't have an `envPath`.

**Comment by hbenl** [comment]
> I agree that in the workspace setting it should fail. But if set in the user settings, it should be liberal in not failing if it does not exist (it could just log a warning).

OK, that is a good idea. I have implemented this in version 2.5.0.

**Comment by alexkli** [comment]
Works like a charm!
